### PR TITLE
feat: [HL-19] Add and configure FormTextField

### PR DESCRIPTION
**Pull request by ahmednasr8898**
Description:
1- add UITextField Style in extension.
2- add FormTextField Reusable view.

### CODE REVIEW

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        super.init(frame: frame)
+        initView()
+    }
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        initView()
+    }
+    
+    private func initView() {
+        Bundle.main.loadNibNamed(FormTextFieldResources.nibName, owner: self, options: nil)
```
Comment (ahmdyasser) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: You can use this handy extension to load nibs
``` swift
extension UIView {
    class func loadNib<T: UIView>() -> T {
      return Bundle(for: T.self).loadNibNamed(String(describing: T.self), owner: nil, options: nil)!.first as! T
    }
}
````

and your code will be like this 
``` swift

let view: UIView = .loadNib()

``` 

:warning: Just be careful, for this extension to work, you **must** name the XIB file exactly like the view class name.

Also an improvement for this code, you can create a factory to create the view for you.

``` swift
class FormTextFieldFactory {
  func createView(title: String) -> FormTextField {
    var view: FormTextField = .loadNib
    view.setTitle(title) // just an example
    return view
  }
}

let factory = FormTextFieldFactory()
factory.createView(title: "Awesome") // FormTextField view
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 2]
```diff
@@ -6,11 +6,14 @@ class FormTextField: UIView {
     // MARK: - Outlets
     //
     
-    @IBOutlet weak private(set) var contentView: UIView!
```
Comment (ahmdyasser) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: Why changed from `private(set)` to `private`? 
How will you test the outlets if they are private?

File: Healthy/Classes/Extensions/UITextField+Style.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        self.rightView = paddingView
+        self.rightViewMode = .always
+    }
+}
+
+// MARK: - Constants
+//
+private extension UITextField {
+    enum Constants {
+        static let defaultCornerRadius: CGFloat = 10.0
```
Comment (ahmdyasser) [inline Healthy/Classes/Extensions/UITextField+Style.swift:50]: Why creating constants as an enum inside a private extension? 
If I have global constants I would create constants class with nested scopes. 
Also if you have a design system this will be redundant work to do.

**Review by ahmdyasser — CHANGES_REQUESTED** [review]

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 2 above)
Comment (ahmednasr8898) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: even don't access the outlets from view controller just working with available functions.

File: Healthy/Classes/Extensions/UITextField+Style.swift (hunk 3 above)
Comment (ahmednasr8898) [reply Healthy/Classes/Extensions/UITextField+Style.swift:50]: this constants just related to UITextField.
what is the best solution you mean.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 1 above)
Comment (ahmednasr8898) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: okay, i will read more about factory and apply on this view.
but i have a question! if i don't using this view by programmatically just using by outlet in this way factory is useful ???

Comment (ahmdyasser) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: Factory pattern makes creating objects easier for you, you pass the changing values to the factory and it hides from you all the mess like loading from nib in our case.
Check this amazing resource  https://refactoring.guru/design-patterns/factory-comparison

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 2 above)
Comment (ahmdyasser) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: Yes but we need to access them for testing.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextFieldResources.swift [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+import Foundation
+
+
+enum FormTextFieldResources {
+    static let nibName: String = "FormTextField"
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextFieldResources.swift]: We don't use this pattern. Each view should have a reusable identifier. 

```swift
// MARK: UIView+Reusable Identifier
extension UIView { 
    static var reusableIdentifier: String { 
        String("\(Self.self)")
    }
}
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    func setError(_ errorText: String) {
+        self.errorLabel.isHidden = false
+        self.errorLabel.text = errorText
+    }
+}
+
+// MARK: - Private Handlers
+//
+
+extension FormTextField {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
//
extension FormTextField {
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    private func setupDesign() {
+        titleLabel.applyStyle(.textFieldTitleLabel)
+        inputTextField.applyTextFieldStyle(.primary)
+        
+        // waiting for design of error label style
+        //MARK: - feat: [HL-4] Add UI label styling new design
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
        // TODO: - feat: [HL-4] Add UI label styling new design
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+// MARK: - Private Handlers
+//
+
+extension FormTextField {
+    private func setupDesign() {
+        titleLabel.applyStyle(.textFieldTitleLabel)
+        inputTextField.applyTextFieldStyle(.primary)
+        
+        // waiting for design of error label style
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
        // Waiting for the design of the error label style
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 8]
```diff
@@ -0,0 +1,78 @@
+
+import UIKit
+
+class FormTextField: UIView {
+    
+    // MARK: - Outlets
+    //
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    // MARK: Outlets
```
We don't usually use MARK with separator inside the context

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    
+    @IBOutlet weak private var contentView: UIView!
+    @IBOutlet weak private var titleLabel: UILabel!
+    @IBOutlet weak private var inputTextField: UITextField!
+    @IBOutlet weak private var errorLabel: UILabel!
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    @IBOutlet private weak var contentView: UIView!
    @IBOutlet private weak var titleLabel: UILabel!
    @IBOutlet private weak var inputTextField: UITextField!
    @IBOutlet private weak var errorLabel: UILabel!
```

Access modifier comes before the reference type.

Comment (ahmdmhasn) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: Please make sure to install swift lint until we integrate the CI

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    
+    // MARK: - Properties
+    //
+    var text: ((String)->())?
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    // MARK: Properties

    private var onChange: (String) -> () =  { _ in }
```
Text is a noun, and this is a closure. I would suggest renaming it to `onChange` and add a setter method. 

Example: 
```swift
func onChange(_ onChange: @escaping (String) -> Void { 
    self.onChange = onChange
}
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    
+    // MARK: - init
+    //
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    // MARK: Init

```
Commented message should follow sentence case (Starts with a capital letter)

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        super.init(frame: frame)
+        initView()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        initView()
+    }
+    
+    private func initView() {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift:30]:
```suggestion
    // MARK: Configurations

    private func initView() {
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 1 above)
Comment (ahmdmhasn) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: It's against the approach of using nib files in the project. The factory would be great if we're following programmatic views.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+        self.errorLabel.isHidden = false
+        self.errorLabel.text = errorText
+    }
+}
+
+// MARK: - Private Handlers
+//
+
+extension FormTextField {
+    private func setupDesign() {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    private func configureLayout() {
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+        titleLabel.applyStyle(.textFieldTitleLabel)
+        inputTextField.applyTextFieldStyle(.primary)
+        
+        // waiting for design of error label style
+        //MARK: - feat: [HL-4] Add UI label styling new design
+        errorLabel.textColor = .red
+    }
+    
+    private func bindToTextFieldsChanges() {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    private func bindTextFieldChanges() {
```
- No need for prepositions 
- We have only one text field, no need for a plural

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+        setupDesign()
+        bindToTextFieldsChanges()
+    }
+}
+
+// MARK: - configure
+//
+
+extension FormTextField {
+    func configure(title: String, placeholder: String?, type: UIKeyboardType) {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: What if we need to configure the keyboard type only? placeholder? Let's create a computed property for each property instead.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        setupDesign()
+        bindToTextFieldsChanges()
+    }
+}
+
+// MARK: - configure
+//
+
+extension FormTextField {
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
//
extension FormTextField {
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+    func configure(title: String, placeholder: String?, type: UIKeyboardType) {
+        self.titleLabel.text = title
+        self.inputTextField.placeholder = placeholder
+        self.inputTextField.keyboardType = type
+    }
+    
+    func setError(_ errorText: String) {
+        self.errorLabel.isHidden = false
+        self.errorLabel.text = errorText
+    }
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    func setError(_ errorText: String) {
        self.errorLabel.isHidden = false
        self.errorLabel.text = errorText
    }
```
You also need to handle the case when we need to hide the error message

File: Healthy/Classes/Generated/UIImage.Generated.swift [hunk 18]
```diff
@@ -6,7 +6,7 @@ import UIKit.UIImage
 // swiftlint:disable force_unwrapping
 extension UIImage {
 
-    static var accentColor: UIImage {
-      UIImage(named: "AccentColor")!
+    static var accentColor : UIImage {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Generated/UIImage.Generated.swift]: Please revert the changes to this file.

File: Healthy/Classes/Extensions/UITextField+Style.swift (hunk 3 above)
Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UITextField+Style.swift:50]: I agree with Nasr in this case, since the scope of this variable is limited to this type. Unless we have a design system, we need to keep this that way.

File: Healthy/Classes/Extensions/UITextField+Style.swift [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(equalToConstant: Constants.defaultHeight),
+        ])
+        
+        layer.cornerRadius = Constants.defaultCornerRadius
+        layer.borderWidth = Constants.defaultBorderWidth
+        layer.borderColor = Constants.defaultBorderColor.cgColor
+        
+        setLeftPaddingPoints()
+        setRightPaddingPoints()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UITextField+Style.swift:26]: Why?

File: Healthy/Classes/Extensions/UITextField+Style.swift [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+        
+        layer.masksToBounds = true
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UITextField+Style.swift]: ?

File: Healthy/Classes/Extensions/UITextField+Style.swift [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+    }
+}
+
+// MARK: - set right / left padding to text field
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UITextField+Style.swift]: We might need to remove this based on the discussion above

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+        </placeholder>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <view contentMode="scaleToFill" id="iN0-l3-epB">
+            <rect key="frame" x="0.0" y="0.0" width="604" height="147"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <subviews>
+                <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="f0b-KT-Zc1">
+                    <rect key="frame" x="0.0" y="0.0" width="604" height="147"/>
+                    <subviews>
+                        <stackView opaque="NO" contentMode="scaleToFill" axis="vertical" spacing="8" translatesAutoresizingMaskIntoConstraints="NO" id="kpw-hB-vQE">
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib]: Why adding the view -- view -- stack view? 
The inner view is not needed

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+                                    <nil key="highlightedColor"/>
+                                </label>
+                            </subviews>
+                        </stackView>
+                    </subviews>
+                    <color key="backgroundColor" systemColor="systemBackgroundColor"/>
+                    <constraints>
+                        <constraint firstItem="kpw-hB-vQE" firstAttribute="leading" secondItem="f0b-KT-Zc1" secondAttribute="leading" id="BV2-PC-nWa"/>
+                        <constraint firstItem="kpw-hB-vQE" firstAttribute="top" secondItem="f0b-KT-Zc1" secondAttribute="top" constant="12" id="WA4-Rw-rgb"/>
+                        <constraint firstAttribute="trailing" secondItem="kpw-hB-vQE" secondAttribute="trailing" id="s96-Y7-ekV"/>
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib]: Bottom constraint is missing

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+                                    <nil key="textColor"/>
+                                    <nil key="highlightedColor"/>
+                                </label>
+                            </subviews>
+                        </stackView>
+                    </subviews>
+                    <color key="backgroundColor" systemColor="systemBackgroundColor"/>
+                    <constraints>
+                        <constraint firstItem="kpw-hB-vQE" firstAttribute="leading" secondItem="f0b-KT-Zc1" secondAttribute="leading" id="BV2-PC-nWa"/>
+                        <constraint firstItem="kpw-hB-vQE" firstAttribute="top" secondItem="f0b-KT-Zc1" secondAttribute="top" constant="12" id="WA4-Rw-rgb"/>
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.xib]:
```suggestion
                        <constraint firstItem="kpw-hB-vQE" firstAttribute="top" secondItem="f0b-KT-Zc1" secondAttribute="top" id="WA4-Rw-rgb"/>
```
Let's add the reusable views without spacing and let the parents do that if needed.

**Review by ahmdmhasn — COMMENTED** [review]
The branch name should follow snack-case. 

Good job 🙏🏼

**Comment by ahmdmhasn** [comment]
Installing swift lint is mandatory, the code should not compile.

File: Healthy/Classes/Extensions/UITextField+Style.swift [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+
+// MARK: - Constants
+//
+private extension UITextField {
+    enum Constants {
+        static let defaultCornerRadius: CGFloat = 10.0
+        static let defaultHeight: CGFloat = 55.0
+        static let defaultPadding: CGFloat = 12.0
+        static let defaultBorderWidth: CGFloat = 1.5
+        static let defaultBorderColor: UIColor = UIColor(red: 0.851, green: 0.851, blue: 0.851, alpha: 1)
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UITextField+Style.swift:55]: TODO for updating the colors here as well

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 17 above)
Comment (ahmednasr8898) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]: I hide the error text when he starts writing again.

File: Healthy/Classes/Extensions/UITextField+Style.swift (hunk 19 above)
Comment (ahmednasr8898) [reply Healthy/Classes/Extensions/UITextField+Style.swift:26]: because text field in design have padding. so i added this functions.

File: Healthy/Classes/Extensions/UITextField+Style.swift (hunk 20 above)
Comment (ahmednasr8898) [reply Healthy/Classes/Extensions/UITextField+Style.swift]: To remove the extra edges in text field.

Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UITextField+Style.swift]: This is a workaround we may not need. Updating the textField `borderStyle` to .none would be enough IMHO.

File: Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+//  UIView+ReusableIdentifier.swift
+//  Healthy
+//
+//  Created by Ahmed Nasr on 25/05/2023.
+//
+
+import UIKit
+
+
+// MARK: UIView+Reusable Identifier
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift]:
```suggestion
import UIKit

// MARK: UIView+Reusable Identifier
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+// MARK: Internal Handlers
+
+extension FormTextField {
+    func setError(_ errorText: String) {
+        self.errorLabel.text = ""
+        self.errorLabel.text = errorText
+    }
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    func setError(_ errorText: String?) {
        self.errorLabel.text = errorText
    }
```

This should be computed property as well.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 28]
```diff
@@ -0,0 +1,90 @@
+
+import UIKit
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
import UIKit
```

File: Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+
+extension UIView {
+    static var reusableIdentifier: String {
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift:7]:
```suggestion
    /// returns a String value, which is the class name of the UIView subclass that it is called on.
    static var reusableIdentifier: String {
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 30]
```diff
@@ -0,0 +1,90 @@
...
+        didSet {
+            self.inputTextField.keyboardType = keyboardType
+        }
+    }
+    
+    var error: String = "" {
+        didSet {
+            self.errorLabel.text = error
+        }
+    }
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    var title: String = "" {
        didSet {
            self.titleLabel.text = title
        }
    }
    
    var placeholder: String = "" {
        didSet {
            self.inputTextField.placeholder = placeholder
        }
    }
    
    var keyboardType: UIKeyboardType = .default {
        didSet {
            self.inputTextField.keyboardType = keyboardType
        }
    }
    
    var error: String = "" {
        didSet {
            self.errorLabel.text = error
        }
    }
```

- All of these should be moved to a separate extension.
- They can act as setters/getters only without having any value to have only on source of truth (The views)

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift (hunk 12 above)
Comment (ahmdmhasn) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift:30]: Not applied?

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 31]
```diff
@@ -0,0 +1,90 @@
...
+        inputTextField.applyTextFieldStyle(.primary)
+        
+        // Waiting for the design of the error label style
+        // TODO: - feat: [HL-4] Add UI label styling new design
+        errorLabel.textColor = .red
+    }
+    
+    private func bindTextFieldChanges() {
+        inputTextField.addTarget(self, action: #selector(textEditingChanged), for: .editingChanged)
+    }
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
    private func configureTextFieldObserver() {
        inputTextField.addTarget(self, action: #selector(textEditingChanged), for: .editingChanged)
    }
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 32]
```diff
@@ -0,0 +1,90 @@
...
+        Bundle.main.loadNibNamed(FormTextField.reusableIdentifier, owner: self, options: nil)
+        addSubview(contentView)
+        contentView.frame = self.bounds
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        configureLayout()
+        bindTextFieldChanges()
+    }
+}
+
+// MARK: Internal Handlers
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
// MARK: Handlers
```
We have plans to apply modularity in the app, we mostly will forget to update this title to Public instead of Internal. This is why we may try to avoid namings like this.

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 33]
```diff
@@ -0,0 +1,90 @@
...
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        initView()
+    }
+    
+    private func initView() {
+        Bundle.main.loadNibNamed(FormTextField.reusableIdentifier, owner: self, options: nil)
+        addSubview(contentView)
+        contentView.frame = self.bounds
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift:40]: I would suggest moving this code to a separate method, ex. `loadContentViewFromNib`

Comment (ahmdmhasn) [reply Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift:40]: Remember SOC principle

File: Healthy/Classes/Extensions/UITextField+Style.swift (hunk 19 above)
Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UITextField+Style.swift:26]: Please document this as a workaround 
```suggestion
        // Workaround for adding leading & training spaces
        setLeftPaddingPoints()
        setRightPaddingPoints()
```

**Comment by ahmdmhasn** [comment]
@ahmednasr8898 Hello 👀

File: Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift [hunk 34]
```diff
@@ -0,0 +1,11 @@
+import UIKit
+
+
+// MARK: UIView+Reusable Identifier
+
+extension UIView {
+    /// returns a String value, which is the class name of the UIView subclass that it is called on.
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIView+ReusableIdentifier.swift]:
```suggestion
    /// Returns a String value, which is the class name of the UIView subclass that it is called on.
```

File: Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift [hunk 35]
```diff
@@ -0,0 +1,114 @@
...
+    var error: String {
+        get {
+            return self.errorLabel.text ?? ""
+        }
+        
+        set {
+            self.errorLabel.text = newValue
+        }
+    }
+}
```
Comment (ahmdmhasn) [inline Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift]:
```suggestion
// MARK: Properties

extension FormTextField {
    var title: String {
        get {
            return self.titleLabel.text ?? ""
        }
        set {
            self.titleLabel.text = newValue
        }
    }
    
    var placeholder: String {
        get {
            return self.inputTextField.placeholder ?? ""
        }
        set {
            self.inputTextField.placeholder = newValue
        }
    }
    
    var keyboardType: UIKeyboardType {
        get {
            return self.inputTextField.keyboardType
        }
        set {
            self.inputTextField.keyboardType = newValue
        }
    }
    
    var error: String {
        get {
            return self.errorLabel.text ?? ""
        }
        set {
            self.errorLabel.text = newValue
        }
    }
}
```

**Comment by ahmdmhasn** [comment]
Merging the ticket to unblock the team. TODOs will be addressed in https://motoon-eg.atlassian.net/browse/HL-46.

@ahmednasr8898  Please acknowledge the message and LMK if you have any questions.
